source/common/network/listen_socket_impl.cc

```diff
@@ -60,6 +60,35 @@ TcpListenSocket::TcpListenSocket(int fd, const Address::InstanceConstSharedPtr&
   setListenSocketOptions(options);
 }
 
+UdpListenSocket::UdpListenSocket(const Address::InstanceConstSharedPtr& address,
+                                 const Network::Socket::OptionsSharedPtr& options,
+                                 bool bind_to_port)
+    : ListenSocketImpl(address->socket(Address::SocketType::Datagram), address) {
+  RELEASE_ASSERT(fd_ != -1, "");
+
+  // TODO(htuch): This might benefit from moving to SocketOptionImpl.
+  // TODO(cmluciano): Analyze special options to be enabled/disabled for UDP
+  int on = 1;
+  int rc = setsockopt(fd_, SOL_SOCKET, SO_REUSEADDR, &on, sizeof(on));
+  RELEASE_ASSERT(rc != -1, fmt::format("failed to set UDP SO_REUSEADDR socket option"));
+  rc = setsockopt(fd_, SOL_SOCKET, SO_SNDBUF, &on, sizeof(on));
```

Comment:
Maybe I'm misreading, but I think the SO_SNDBUF and SO_RCVBUF sockopts here might be configuring the socket to have send and receive buffers of size 1 (byte).  An alternative might be to leave out these setsockopt calls so the socket uses system default buffer sizes, and if someone wants to tune the buffer sizes, they could do by specifying socket_options in the config for the listener?

Reply:
Great catch, this was a poor choice on my part. I will leave these parts out for now and leave a note to shift these into a factory for later.

Reply:
Sounds good, thanks!  BTW, let me know if you'd like an extra pair of hands for this, I'd be happy to help in whatever way is constructive. (Also fully understand if that just complicates things and you'd prefer not--just wanted to put the offer out there.)

Reply:
Happy to have help if I get stuck, will ping you on Slack early next week.